Code/Framework/AzToolsFramework/AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/Event.h>
+#include <AzFramework/Viewport/ViewportId.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiRequestBus.h>
+
+namespace AzToolsFramework
+{
+    //! Enumeration of each viewport editor mode.
+    enum class ViewportEditorMode : AZ::u8
+    {
+        Default,
+        Component,
+        Focus,
+        Pick
+    };
```

Comment:
question: Just had a thought that should this be a mask? (as we can be in more than one mode at once in certain situations) or is this handled in another way?

Reply:
Correct, we can be in more than one mode. A mask could be used instead but I think we should instead move over to the model of having the tracked modes added at runtime (but not in this PR).